hoodie-client/src/test/java/com/uber/hoodie/io/TestHoodieCommitArchiveLog.java

```diff
@@ -42,23 +42,56 @@
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.DistributedFileSystem;
+import org.apache.hadoop.hdfs.MiniDFSCluster;
+import org.junit.AfterClass;
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 
 public class TestHoodieCommitArchiveLog {
 
   private String basePath;
-  private FileSystem fs;
   private Configuration hadoopConf;
+  //NOTE : Be careful in using DFS (FileSystem.class) vs LocalFs(RawLocalFileSystem.class)
+  //The implementation and gurantees of many API's differ, for example check rename(src,dst)
+  // We need to use DFS here instead of LocalFs since the FsDataInputStream.getWrappedStream() returns a FsDataInputStream
```

Comment:
Another extremely tedious thing to track down, wish LocalFS and DFS had same contracts :(